pkg/terraform/hcl/backend.go

```diff
@@ -11,25 +11,26 @@ import (
 )
 
 type BackendBlock struct {
-	Name          string   `hcl:"name,label"`
-	Path          string   `hcl:"path,optional"`
-	WorkspaceDir  string   `hcl:"workspace_dir,optional"`
-	Bucket        string   `hcl:"bucket,optional"`
-	Key           string   `hcl:"key,optional"`
-	Region        string   `hcl:"region,optional"`
-	Prefix        string   `hcl:"prefix,optional"`
-	ContainerName string   `hcl:"container_name,optional"`
-	Remain        hcl.Body `hcl:",remain"`
+	Name               string   `hcl:"name,label"`
+	Path               string   `hcl:"path,optional"`
+	WorkspaceDir       string   `hcl:"workspace_dir,optional"`
+	Bucket             string   `hcl:"bucket,optional"`
+	Key                string   `hcl:"key,optional"`
+	Region             string   `hcl:"region,optional"`
+	Prefix             string   `hcl:"prefix,optional"`
+	ContainerName      string   `hcl:"container_name,optional"`
+	WorkspaceKeyPrefix string   `hcl:"workspace_key_prefix,optional"`
+	Remain             hcl.Body `hcl:",remain"`
 }
 
-func (b BackendBlock) SupplierConfig() *config.SupplierConfig {
+func (b BackendBlock) SupplierConfig(workspace string) *config.SupplierConfig {
 	switch b.Name {
 	case "local":
 		return b.parseLocalBackend()
 	case "s3":
-		return b.parseS3Backend()
+		return b.parseS3Backend(workspace)
```

Comment:
Using the workspace in the path of the state is used only for S3 and GCS, not azure ?

Reply:
I tested it and indeed, it is also used by azurerm!